tests/v2/validation/auth/userretention/userretention.go

```diff
@@ -0,0 +1,166 @@
+package userretention
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/rancher/machine/libmachine/log"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/tests/v2/actions/auth"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/kubeapi/rbac"
+	"github.com/rancher/shepherd/extensions/users"
+	rbacv1 "k8s.io/api/rbac/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	disableInactiveUserAfter = "disable-inactive-user-after"
+	deleteInactiveUserAfter  = "delete-inactive-user-after"
+	userRetentionCron        = "user-retention-cron"
+	userRetentionDryRun      = "user-retention-dry-run"
+	lastloginLabel           = "cattle.io/last-login"
+	defaultWaitDuration      = 70 * time.Second
+	pollInterval             = 10 * time.Second
+	isActive                 = true
+	isInActive               = false
+	webhookErrorMessage      = "admission webhook \"rancher.cattle.io.settings.management.cattle.io\" denied the request: value: Invalid value:"
+)
+
+func setupUserRetentionSettings(client *rancher.Client, disableAfterValue string, deleteAfterValue string, userRetentionCronValue string, dryRunValue string) error {
+	log.Info("Setting up user retention settings TO : ")
+	log.Infof("DisableAfterValue as %s, DeleteAfterValue as %s, UserRetentionCronValue as %s, DryRunValue as %s", disableAfterValue, deleteAfterValue, userRetentionCronValue, dryRunValue)
+	err := updateUserRetentionSettings(client, disableInactiveUserAfter, disableAfterValue)
+	if err != nil {
+		return fmt.Errorf("failed to update disable-inactive-user-after setting: %w", err)
+	}
+	err = updateUserRetentionSettings(client, deleteInactiveUserAfter, deleteAfterValue)
+	if err != nil {
+		return fmt.Errorf("failed to update delete-inactive-user-after setting: %w", err)
+	}
+
+	err = updateUserRetentionSettings(client, userRetentionCron, userRetentionCronValue)
+	if err != nil {
+		return fmt.Errorf("failed to update user-retention-cron setting: %w", err)
+	}
+	err = updateUserRetentionSettings(client, userRetentionDryRun, dryRunValue)
```

Comment:
we are using the same function multiple times here.. why not just call updateUserRetentionSettings from the test instead of using `setupUserRetentionSettings`?  I ask because this is too many iterations for a single setting to be updated.

Reply:
I did not fully get you this comment but the test will setup the data Initially by using setupUserRetentionSettings and the update function(updateUserRetentionSettings) will use to update individual settings.